File: solutions/aoc2021_day2.py
def part1(arr):
    x = 0
    y = 0
    for command in arr:
        a, b = command.split(" ")
        direction = a
        magnitude = int(b)
        if direction == "forward":
            x += magnitude
        if direction == "down":
            y += magnitude
        if direction == "up":
            y -= magnitude
    ans = x * y
    print("part1: ", ans)

def part2(arr):
    x = 0
    y = 0
    aim = 0
    for command in arr:
        a, b = command.split(" ")
        direction = a
        magnitude = int(b)
        if direction == "forward":
            x += magnitude
            y += aim * magnitude
        if direction == "down":
            aim += magnitude
        if direction == "up":
            aim -= magnitude
    ans = x * y
    print("part2: ", ans)

with open("aoc2021_day2.txt") as f:
    data = f.read()
    arr = [line for line in data.splitlines()]

print("day2")
part1(arr)
part2(arr)

File: solutions/aoc2021_day2.js
var fs = require('fs');

function part1(pi) {
    var arr = pi.split('\n').map(String);
    var x = 0;
    var y = 0;
    for(var command of arr) {
        var comp = command.split(" ");
        var direction = comp[0];
        var magnitude = parseInt(comp[1]);
        if (direction === "forward") {
            x += magnitude;
        }
        else if (direction === "down") {
            y += magnitude;
        }
        else if (direction === "up") {
            y -= magnitude;
        }
    }
    var ans = x * y;
    console.log(`part1: ${ans}`);
}

function part2(pi) {
    var arr = pi.split('\n').map(String);
    var x = 0;
    var y = 0;
    var aim = 0;
    for (var command of arr) {
        var comp = command.split(" ");
        var direction = comp[0];
        var magnitude = parseInt(comp[1]);
        if (direction === "forward") {
            x += magnitude;
            y += aim * magnitude;
        }
        else if (direction === "down") {
            aim += magnitude;
        }
        else if (direction === "up") {
            aim -= magnitude;
        }
    }
    var ans = x * y;
    console.log(`part2: ${ans}`);
}

console.log("day2");
var f = fs.readFileSync('aoc2021_day2.txt', 'utf8');
var pi = f.toString();
part1(pi);
part2(pi);